app/src/main/java/com/duckduckgo/app/browser/senseofprotection/SenseOfProtectionExperimentImpl.kt

```diff
@@ -0,0 +1,176 @@
+    private val pixel: Pixel,
+) : SenseOfProtectionExperiment {
+
+    init {
+        // enrol users in the existing user experiment if they are not already enrolled in the new user experiment
+        if (userBrowserProperties.daysSinceInstalled() > EXISTING_USER_DAY_COUNT_THRESHOLD) {
+            if (isNotEnrolledInNewUserExperiment()) {
+                isExistingUserExperimentEnabled(cohortName = MODIFIED_CONTROL)
+            }
+        }
+    }
+
+    override fun enrolUserInNewExperimentIfEligible(): Boolean {
+        return if (userBrowserProperties.daysSinceInstalled() <= EXISTING_USER_DAY_COUNT_THRESHOLD) {
+            isNewUserExperimentEnabled(cohortName = MODIFIED_CONTROL)
+        } else {
+            false
+        }
+    }
+
+    override fun isUserEnrolledInVariant2CohortAndExperimentEnabled(): Boolean =
+        getNewUserExperimentCohortName() == VARIANT_2.cohortName && isNewUserExperimentEnabled(VARIANT_2) ||
+        getExistingUserExperimentCohortName() == VARIANT_2.cohortName && isExistingUserExperimentEnabled(VARIANT_2)
+
+    override fun isMemberOfVariant2Cohort(): Boolean =
+        senseOfProtectionToggles.senseOfProtectionNewUserExperimentApr25().getCohort()?.name == SenseOfProtectionToggles.Cohorts.VARIANT_2.cohortName ||
+            senseOfProtectionToggles.senseOfProtectionExistingUserExperimentApr25().getCohort()?.name == SenseOfProtectionToggles.Cohorts.VARIANT_2.cohortName
+
+    override fun getTabManagerPixelParams(): Map<String, String> {
+        return when {
+            isEnrolledInNewUserExperiment() -> {
+                mapOf(
+                    "cohort" to (getNewUserExperimentCohortName() ?: ""),
+                    "experiment" to getNewUserExperimentName(),
+                )
+            }
+
+            isEnrolledInExistingUserExperiment() -> {
+                mapOf(
+                    "cohort" to (getExistingUserExperimentCohortName() ?: ""),
+                    "experiment" to getExistingUserExperimentName(),
+                )
+            }
+
+            else -> emptyMap()
+        }
+    }
+
+    override fun firePrivacyDashboardClickedPixelIfInExperiment() {
+        appCoroutineScope.launch(dispatcherProvider.io()) {
+            if (isUserEnrolledInAVariantAndExperimentEnabled()) {
+                senseOfProtectionPixelsPlugin.getPrivacyDashboardClickedMetric()?.fire()
+            }
+        }
+    }
+
+    override fun isUserEnrolledInModifiedControlCohortAndExperimentEnabled(): Boolean =
+        getNewUserExperimentCohortName() == MODIFIED_CONTROL.cohortName && isNewUserExperimentEnabled(MODIFIED_CONTROL) ||
+            getExistingUserExperimentCohortName() == MODIFIED_CONTROL.cohortName && isExistingUserExperimentEnabled(MODIFIED_CONTROL)
+
+    // TODO this _might_ be able to be simplified like the isUserEnrolledInModifiedControlCohortAndExperimentEnabled function
+    override fun isUserEnrolledInAVariantAndExperimentEnabled(): Boolean {
```

Comment:
We just added new APIs to facilitate this.  

`fun isEnrolledAndEnabled(cohort: CohortName): Boolean`

This will return true if the user is enrolled in the cohort and the experiment is enabled.